cmake/ExtendBuildTypes.cmake

```diff
@@ -19,7 +19,7 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
     add_compile_options(-Wshadow=local)
   endif()
 else()
-  add_compile_options(-Wshadow -Wno-c99-designator)
+  add_compile_options(-Wshadow -Wno-c99-designator -fsized-deallocation)
```

Comment:
I can't remember what the defaults are across compilers. Thanks for adding this in. I guess we could also just add it for all versions / compilers ?

Reply:
this is has been the default for gcc for a long time, only necessary for clang.